core/model/modx/processors/search/search.class.php

```diff
@@ -172,13 +172,14 @@ public function searchResources()
         ));
         $c->sortby('createdon', 'DESC');
 
-        $c->limit($this->maxResults);
+        $c->limit($this->modx->getOption('maxsearchresults',null,$maxResults));
```

Comment:
Instead of overriding for each "case", you could change the attribute (`$this->maxResults`) in the process method, or the like